### PR TITLE
CLC-5906, new tool for /ccadmin: Reload YAML settings

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/CLC-5923
Before I drop in the core implementation of "Reload YAML settings" I've shuffled the chairs and updated /ccadmin.  With this PR, we have the *refresh_logging* logic as a placeholder. It will be replaced in subsequent PR.

Minor refactoring of spec, too.

### CODE REVIEW

File: lib/calcentral_config.rb [hunk 1]
```diff
@@ -13,6 +13,25 @@ def load_settings
...
+      old_level_name = Log4r::LNAMES[old_level]
+      new_level_name = Log4r::LNAMES[new_level]
+      Rails.logger.level = new_level
+      Rails.logger.warn "Rails.logger.level changed (old -> new): #{old_level_name} -> #{new_level_name}"
+    else
+      Rails.logger.warn "Do nothing. Log levels have not changed (old -> new): #{old_level} -> #{new_level}"
+    end
+    valid
+  end
+
```
Comment (johncrossman) [inline lib/calcentral_config.rb]: Implementation of the method above will change in subsequent PR.

**Comment by johncrossman** [comment]
1 out of 2 Travis.

File: spec/controllers/yaml_settings_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
+describe YamlSettingsController do
+
+  before(:each) do
+    @user_id = rand(99999).to_s
+    session['user_id'] = @user_id
+    expect(Rails.env).to receive(:production?).at_least(1).times.and_return true
+    expect(User::Auth).to receive(:where).and_return [user_auth]
+    # allow(Settings.features).to receive(:reauthentication).and_return true
```
Comment (paulkerschen) [inline spec/controllers/yaml_settings_controller_spec.rb]: @johncrossman Will this commented-out line be used in future work, or should it go?

**Comment by paulkerschen** [comment]
@johncrossman Aside from one stray line this all looks good to me, but @raydavis might want to look too, since I expect he knows this part of the code better than I do.

File: app/controllers/yaml_settings_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class YamlSettingsController < ApplicationController
+  include ClassLogger
+
+  respond_to :json
+
+  def reload
+    authorize(current_user, :can_clear_cache?)
```
Comment (raydavis) [inline app/controllers/yaml_settings_controller.rb]: Scares me a bit to overload the "can_clear_cache" permission like this -- the two actions are conceptually pretty different, even though they do go together as a practical matter.

**Comment by johncrossman** [comment]
Thanks @paulkerschen  I had noticed that stray line and committed a removal. As happens, I forgot to push. It is done.

File: app/controllers/yaml_settings_controller.rb (hunk 3 above)
Comment (johncrossman) [inline app/controllers/yaml_settings_controller.rb]: I thought you might say that. :)  I didn't want to over litter *AuthenticationStatePolicy* but you are right. I'll fix.

**Comment by paulkerschen** [comment]
Onward!

**Pull request merged**